File: apps/python-sdk/firecrawl/v2/methods/usage.py
from ..utils import HttpClient, handle_response_error
from ..types import ConcurrencyCheck, CreditUsage, TokenUsage


def get_concurrency(client: HttpClient) -> ConcurrencyCheck:
    resp = client.get("/v2/concurrency-check")
    if not resp.ok:
        handle_response_error(resp, "get concurrency")
    body = resp.json()
    if not body.get("success"):
        raise Exception(body.get("error", "Unknown error"))
    data = body.get("data", body)
    return ConcurrencyCheck(
        concurrency=data.get("concurrency"),
        max_concurrency=data.get("maxConcurrency", data.get("max_concurrency")),
    )


def get_credit_usage(client: HttpClient) -> CreditUsage:
    resp = client.get("/v2/team/credit-usage")
    if not resp.ok:
        handle_response_error(resp, "get credit usage")
    body = resp.json()
    if not body.get("success"):
        raise Exception(body.get("error", "Unknown error"))
    data = body.get("data", body)
    return CreditUsage(remaining_credits=data.get("remainingCredits", data.get("remaining_credits", 0)))


def get_token_usage(client: HttpClient) -> TokenUsage:
    resp = client.get("/v2/team/token-usage")
    if not resp.ok:
        handle_response_error(resp, "get token usage")
    body = resp.json()
    if not body.get("success"):
        raise Exception(body.get("error", "Unknown error"))
    data = body.get("data", body)
    return TokenUsage(
        remaining_tokens=data.get("remainingTokens", 0)
    )

File: apps/js-sdk/firecrawl/src/v2/methods/usage.ts
import type { ConcurrencyCheck, CreditUsage, TokenUsage } from "../types";
import { HttpClient } from "../utils/httpClient";
import { normalizeAxiosError, throwForBadResponse } from "../utils/errorHandler";

export async function getConcurrency(http: HttpClient): Promise<ConcurrencyCheck> {
  try {
    const res = await http.get<{ success: boolean; data?: { concurrency: number; maxConcurrency: number } }>("/v2/concurrency-check");
    if (res.status !== 200 || !res.data?.success) throwForBadResponse(res, "get concurrency");
    const d = res.data.data || (res.data as any);
    return { concurrency: d.concurrency, maxConcurrency: d.maxConcurrency ?? d.max_concurrency };
  } catch (err: any) {
    if (err?.isAxiosError) return normalizeAxiosError(err, "get concurrency");
    throw err;
  }
}

export async function getCreditUsage(http: HttpClient): Promise<CreditUsage> {
  try {
    const res = await http.get<{ success: boolean; data?: { remaining_credits: number } }>("/v2/team/credit-usage");
    if (res.status !== 200 || !res.data?.success) throwForBadResponse(res, "get credit usage");
    const d = res.data.data || (res.data as any);
    return { remaining_credits: d.remaining_credits };
  } catch (err: any) {
    if (err?.isAxiosError) return normalizeAxiosError(err, "get credit usage");
    throw err;
  }
}

export async function getTokenUsage(http: HttpClient): Promise<TokenUsage> {
  try {
    const res = await http.get<{ success: boolean; data?: TokenUsage }>("/v2/team/token-usage");
    if (res.status !== 200 || !res.data?.success) throwForBadResponse(res, "get token usage");
    return (res.data.data || (res.data as any)) as TokenUsage;
  } catch (err: any) {
    if (err?.isAxiosError) return normalizeAxiosError(err, "get token usage");
    throw err;
  }
}

